### PR TITLE
Make sure the JSON logging formatter is using the correct constructor signature.

### LINKED ISSUE #24: Constructor of formatter should be (fmt, datefmt, style)
Currently the `style` parameter is used as `logger_name`.

Because of this, when the formatter is initialized from `.ini` files, the `Logger` field is set to `%`:
https://github.com/python/cpython/blob/97d7e65dfed1d42d40d9bc2f630af56240555f02/Lib/logging/config.py#L112-L119

Maybe, the constructor signature could be:
```py
def __init__(self, fmt=None, datefmt=None, style='%', logger_name='Dockerflow'):
    ...
```

(From https://github.com/mozilla/mozilla-cloud-services-logger/issues/3)

### PR DESCRIPTION
**Pull request by jezdez**
Fix #24.

@leplatrem  r?

### CODE REVIEW

File: src/dockerflow/logging.py [hunk 1]
```diff
@@ -47,8 +48,13 @@ class JsonLogFormatter(logging.Formatter):
         'relativeCreated', 'stack_info', 'thread', 'threadName'
     ))
 
-    def __init__(self, format=None, datefmt=None, logger_name='Dockerflow'):
-        super(JsonLogFormatter, self).__init__(format, datefmt)
+    def __init__(self, fmt=None, datefmt=None, style='%', logger_name='Dockerflow'):
+        parent_init = logging.Formatter.__init__
+        argspec = inspect.getargspec(parent_init)
```
Comment (leplatrem) [inline src/dockerflow/logging.py]: Out of curiosity: in which case would the `logging.Formatter` class have a different signature? (maybe we could add a comment)

File: tests/test_logging.py [hunk 2]
```diff
@@ -20,6 +22,44 @@ def assert_records(records):
...
+    formatter=json
+
+    [formatter_json]
+    class = dockerflow.logging.JsonLogFormatter
+    """)
+    ini_file = tmpdir.join('logging.ini')
+    ini_file.write(ini_content)
+    logging.config.fileConfig(str(ini_file))
+    logging.info('I am logging in mozlog format now! woo hoo!')
+    logger = logging.getLogger('test_logger')
```
Comment (leplatrem) [inline tests/test_logging.py]: Wouldn't it work with the `root` logger? In other words, what do you think about reducing the ini to the bare minimum?

File: tests/test_logging.py [hunk 3]
```diff
@@ -20,6 +22,44 @@ def assert_records(records):
...
+    assert len(logger.handlers) > 0
+    assert logger.handlers[0].formatter.logger_name == 'Dockerflow'
```
Comment (leplatrem) [inline tests/test_logging.py]: \o/

File: src/dockerflow/logging.py (hunk 1 above)
Comment (leplatrem) [reply src/dockerflow/logging.py]: Uups, scratch that. I hadn't read your comment on the issue. Definitely worth adding a comment IMO :)

File: tests/test_logging.py (hunk 2 above)
Comment (jezdez) [reply tests/test_logging.py]: Yeah, I tried it with a root logger as well, lemme configure that.

**Comment by jezdez** [comment]
@leplatrem Okay, done.

**Comment by leplatrem** [comment]
Great :) thanks!

**Pull request merged**